### PR TITLE
PR for openapi-generator-maven-plugin inputSpec -- Allow jar: URLs

### PR DESCRIPTION
**Pull request by allen-ball**
This is a PR in response to https://github.com/OpenAPITools/openapi-generator/issues/10016.  It changes `<inputSpec/>` processing to:

1. Search a classpath of Maven compilation dependencies for resources.
2. Allow URLs of the form jar:jar-file-specific-url!/spec.yaml

In the case that a compilation classpath resource is specified, the resource URL is passed to the swagger-parser OpenAPIV3Parser instead of the `<inputSpec/>` string since the OpenAPIV3Parser does not have access to the Maven compilation classpath.  This requires https://github.com/swagger-api/swagger-parser/issues/1592 and https://github.com/swagger-api/swagger-parser/pull/1593.

modules/openapi-generator-maven-plugin/src/test/java/org/openapitools/codegen/plugin/CodeGenMojoTest.java has been expanded to include tests for resource and URL input specs.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1578,7 +1578,7 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <swagger-core-version>2.1.2</swagger-core-version>
         <swagger-parser-groupid>io.swagger.parser.v3</swagger-parser-groupid>
-        <swagger-parser-version>2.0.26</swagger-parser-version>
+        <swagger-parser-version>2.0.28-SNAPSHOT</swagger-parser-version>
```
Comment (wing328) [inline pom.xml:1581]: I think we will need to wait for the 2.0.28 release before merging this enhancement.

**Comment by setu9760** [comment]
@allen-ball Are you looking to fix the merge conflicts on this MR so it can be be ready to merge ? The dependency spring-parser version is now released.

**Comment by nitish-atomic** [comment]
@allen-ball  are you planning to merge this??

File: pom.xml (hunk 1 above)
Comment (CC007) [reply pom.xml:1581]: It seems that [2.1.12](https://mvnrepository.com/artifact/io.swagger.parser.v3/swagger-parser/2.1.12) has already been released

**Comment by sambhav37** [comment]
When are we planning to merge this ? This is much needed.

**Comment by obfischer** [comment]
@jimschubert Can we support you somehow? This is a top 10 feature, needed very much. Being able to take the spec from classpath would ease the setup of a project as well as sharing specs and lead to an better build structure.

**Comment by parenko** [comment]
@allen-ball is the any chance you will work on this PR?
If not, I've [rebased your branch on master](https://github.com/parenko/openapi-generator/tree/fix-10016), fixed merge conflicts, removed swagger-parser 2.0.28-SNAPSHOT since openapi-generator uses [2.1.19](https://github.com/OpenAPITools/openapi-generator/blob/master/pom.xml#L1255) now, and could create a PR.

**Comment by allen-ball** [comment]
Please go ahead and do so — I haven’t been able to find the time to follow up on this.

Thanks!

— Allen

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator/pull/18576

**Pull request closed without merging**